### PR TITLE
Add sample JS Trace Id Emitter app

### PR DESCRIPTION
**Pull request by KKelvinLo**
**What is the point of this pull request?**
This pull request adds a sample JavaScript app which is able to emit trace Ids in order to run in the testing framework to validate traceIds to AWS X-Ray.

### CODE REVIEW

**Comment by wyTrivail** [comment]
i'd like to have Bhautik to take a look :), @bhautikpip

File: sample-apps/js-sdk-aws-XRay/server.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        response.end('hello world!');
+      }, 2000);
+    });
+  } catch (err) {
+    console.error(err);
+    span.end();
+  }
+}
+
+startServer(8080);
```
Comment (wyTrivail) [inline sample-apps/js-sdk-aws-XRay/server.js]: please follow the guide, to use LISTEN_ADDRESS env var first.  you can have 8080 as the default port

File: sample-apps/js-sdk-aws-XRay/server.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  span.addEvent('invoking handleRequest');
+  try {
+    const body = [];
+    request.on('error', (err) => console.log(err));
+    request.on('data', (chunk) => body.push(chunk));
+    request.on('end', () => {
+      // deliberately sleeping to mock some action.
+      setTimeout(() => {
+        span.end();
+        response.end('hello world!');
```
Comment (wyTrivail) [inline sample-apps/js-sdk-aws-XRay/server.js]: please follow the guide, to return "traceId"

File: sample-apps/js-sdk-aws-XRay/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+
+RUN npm install
+
+COPY . .
+
+EXPOSE 8080
```
Comment (wyTrivail) [inline sample-apps/js-sdk-aws-XRay/Dockerfile]: you might not need this, since you will need to read the env var LISTEN_ADDRESS

**Review by wyTrivail — CHANGES_REQUESTED** [review]

File: sample-apps/js-sdk-aws-XRay/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  "keywords": [
+    "opentelemetry",
+    "http",
+    "tracing"
+  ],
+  "engines": {
+    "node": ">=8"
+  },
+  "dependencies": {
```
Comment (wyTrivail) [inline sample-apps/js-sdk-aws-XRay/package.json:17]: you will need to build this image with the repo: https://github.com/aws-observability/aws-otel-js whenever there's code commit, so i'm thinking how do we put the dependencies here, looks like it's using the official repo instead? 

maybe try to add aws-itel-js as a local lib? https://stackoverflow.com/questions/48944051/package-json-add-local-directory-to-node-modules/48944131

File: sample-apps/js-sdk-aws-XRay/server.js (hunk 1 above)
Comment (KKelvinLo) [reply sample-apps/js-sdk-aws-XRay/server.js]: done

File: sample-apps/js-sdk-aws-XRay/server.js (hunk 2 above)
Comment (KKelvinLo) [reply sample-apps/js-sdk-aws-XRay/server.js]: done

File: sample-apps/js-sdk-aws-XRay/Dockerfile (hunk 3 above)
Comment (KKelvinLo) [reply sample-apps/js-sdk-aws-XRay/Dockerfile]: removed

File: sample-apps/js-sdk-aws-XRay/package.json (hunk 4 above)
Comment (KKelvinLo) [reply sample-apps/js-sdk-aws-XRay/package.json:17]: Will be moving to aws-otel-js repository to resolve dependencies issues

File: sample-apps/js-sdk-aws-XRay/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+
+RUN npm install
+
+COPY . .
+
+CMD [ "node", "server.js" ]
```
Comment (bhautikpip) [inline sample-apps/js-sdk-aws-XRay/Dockerfile:11]: nitpick: add a line here

File: sample-apps/js-sdk-aws-XRay/server.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  function returnTraceIdJson(span) {
+    const traceId = span.context().traceId.toString();
+    const xrayTraceId = "1-" + traceId.substring(0, 8) + "-" + traceId.substring(8);
+    const traceIdJson = { "traceId" : xrayTraceId }
+    return traceIdJson;
+  }
+  
+}
+
+startServer(process.env.LISTEN_ADDRESS);
```
Comment (bhautikpip) [inline sample-apps/js-sdk-aws-XRay/server.js:58]: here too!

**Comment by bhautikpip** [comment]
Also, looks like this sample app is hitting one path. What is the purpose of this sample app? How do you want to validate data in validator? If you want to validate various instrumentation then you might consider adding more paths like /aws-sdk-call, /outgoing-http-call to test client side instrumentation.

**Pull request closed without merging**